unyt/dimensions.py

```diff
@@ -193,3 +196,182 @@
 
 for k, v in list(em_dimensions.items()):
     em_dimensions[v] = k
+
+
+def accepts(**arg_units):
+    """Decorator for checking units of function arguments.
+
+    Parameters
+    ----------
+    arg_units: dict
+        Mapping of function arguments to dimensions, of the form 'arg1'=dimension1 etc
+        where ``'arg1'`` etc are the function arguments and ``dimension1`` etc
+        are SI base units (or combination of units), eg. length/time.
```

Comment:
Add a note explaining how to deal with arguments that don't have units. Also note that keyword arguments aren't checked and are instead passed through to the function.